clang-tools-extra/clang-tidy/readability/UseBuiltinLiteralsCheck.cpp

```diff
@@ -0,0 +1,171 @@
+//===--- UseBuiltinLiteralsCheck.cpp - clang-tidy -------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "UseBuiltinLiteralsCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+#include "clang/Lex/Lexer.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang::tidy::readability {
+
+namespace {
+
+static StringRef getRawStringRef(const SourceRange &Range,
+                                 const SourceManager &Sources,
+                                 const LangOptions &LangOpts) {
+  CharSourceRange TextRange = Lexer::getAsCharRange(Range, Sources, LangOpts);
+  return Lexer::getSourceText(TextRange, Sources, LangOpts);
+}
+
+typedef bool (*RuleOnStd)(const LangStandard &LS);
+
+struct Replacement {
+  Replacement(const StringRef &_Seq, const RuleOnStd _Std = NULL)
+      : Seq(_Seq), Std(_Std) {}
```

Comment:
```suggestion
  Replacement(const StringRef &Seq, const RuleOnStd Std = nullptr)
      : Seq(Seq), Std(Std) {}
```

⛔️⛔️⛔️ [**`[lex.name] §3`**](http://eel.is/c++draft/lex.name#3)

> In addition, some identifiers appearing as a token or preprocessing-token are reserved for use by C++ implementations and shall not be used otherwise; no diagnostic is required.
> * Each identifier that contains a double underscore `__` or **begins with an underscore followed by an uppercase letter** is reserved to the implementation **for any use**.
> * Each identifier that begins with an underscore is reserved to the implementation for use as a _name_ in the _global namespace_.